[FIX] product: default pricelist item

nim-odoo opened the pull request:
By default, a pricelist can be created without pricelist item. However,
if a user does not have the rights to manage the pricelist items (or
simply does not want to manage them), the feature is unusable. Indeed,
the user will create a new pricelist (for example, in a new currency),
but this pricelist will not have any item. Therefore, the pricelist will
not apply to anything.

The fix creates a pricelist item by default if none is found. The
pricelist item is the most generic possible: it applies to all products
and uses the list price as a base price.

opw-667607

xmo-odoo commented:
Wouldn't a default value on `item_ids` be simpler and more obvious than overloading `create`? And aside from `compute_price`, all specified value duplicate the default values for pricelist items (which goes to show the defaults aren't bad), it might be better to not do that so the value of the default pricelist item remains in sync with the default value of items in general?

nim-odoo commented:
You are right, a default value is better. I updated the PR, however, that doesn't behave like expected. The new item is indeed added in the pricelist when I click on "Create", but when it comes to saving (call of the `create` method), the default values are gone:
`u'item_ids': [[0, False, {u'min_quantity': 1, u'price': u'0.0 % discount and False surcharge', u'date_start': False, u'name': u'All Products', u'date_end': False}]]`

Moreover, at creation time, the values appear in the tree view of the items, but if I open the form view of the default item, nothing: http://imgur.com/DHh5BDi

Do you have idea idea why?

xmo-odoo commented:
> Do you have idea idea why?

I'm guessing the web client doesn't correctly handle the o2m default? /cc @ged-odoo

nim-odoo commented:
> Is that why you're also fetching the default values for pricelist items?

Yes, my first try was:
```
    _defaults = {
        'active': lambda *a: 1,
        "currency_id": _get_currency,
        'item_ids': [[0, False, {
		'compute_price': 'formula',
	}]],
    }
```
As it was not giving the expected result, I added the call to get the default values. I also added these fields in the tree view, just to be sure. But in the end, the result was the same.

> FWIW if you're fetching default values from a model you should call default_get to correctly resolve all of them from all possible sources, handle inheritance, handle context values, etc...

Thanks for the info
